### PR TITLE
Formalize more string skolems

### PR DESCRIPTION
**Pull request by ajreynol**
This properly formalizes all string skolems used in reduction and preprocessing. This avoids proof checking failures due to non-deterministism when checking steps for these modules.

Fixes https://github.com/cvc5/cvc5-projects/issues/334. Fixes https://github.com/cvc5/cvc5-projects/issues/331.

### CODE REVIEW

File: test/api/proj-issue334.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "api/cpp/cvc5.h"
```
Comment (4tXJ7f) [inline test/api/proj-issue334.cpp:1]: Add boilerplate.

File: src/theory/strings/skolem_cache.h [hunk 2]
```diff
@@ -58,6 +58,10 @@ class SkolemCache
    * preconditions below, e.g. where we are considering a' ++ a = b' ++ b.
    *
    * All skolems assume a and b are strings unless otherwise stated.
+   *
+   * Notice that these identifiers are each syntax sugar for constructing a
+   * purification skolem. It is required for the purposes of proof checking
+   * that this only result in calls to SkolemManager::mkPurifySkolem.
```
Comment (4tXJ7f) [inline src/theory/strings/skolem_cache.h:64]:
```suggestion
   * that this only results in calls to SkolemManager::mkPurifySkolem.
```

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]
Just some minor requests.

File: test/api/proj-issue334.cpp (hunk 1 above)
Comment (ajreynol) [reply test/api/proj-issue334.cpp:1]: What do you mean?

Comment (4tXJ7f) [reply test/api/proj-issue334.cpp:1]: Our usual copyright header.

**Review by 4tXJ7f — APPROVED** [review]
LGTM, thanks for the update.

**Pull request merged**